docs/rhoas/iac_tools_rhoas/README.adoc

```diff
@@ -0,0 +1,602 @@
++
+
+.Example RHOAS provider configuration
+[source,shell]
+----
+$ terraform {
+  required_providers {
+    rhoas = {
+      source = "redhat-developer/rhoas"
+      version = "0.4.0-alpha1"
+    }
+  }
+}
+
+provider "rhoas" {
+  #configuration options
+}
+----
+
+. In your IDE, open a new file and paste the configuration you copied. You can specify configuration options in the provider settings.
++
+
+. Save the file as a Terraform configuration (`main.tf`) file in a directory called `Terraform`.
+. Open a terminal and navigate to the Terraform directory.
++
+[source,shell]
+----
+$ cd Terraform
+----
+. Enter the following command. This command initializes the working directory containing Terraform configuration files and installs any required plug-ins.
++
+[source,shell]
+----
+$ terraform init
+----
+When the Terraform provider has been initialized, you see a confirmation message.
+
+[id="proc-using-terraform_{context}"]
+=== Using the RHOAS Terraform provider
+
+Resources are the most important element in the Terraform language. Each resource block in a Terraform provider describes one or more infrastructure objects. For {product-long-kafka}, such infrastructure objects might include Kafka instances, service accounts, Access Control Lists (ACLs), and topics. The procedure that followS show what resources you can add to your Terraform configuration file to create a Kafka instance and its associated resources such as service accounts and topics.
+
+.Prerequisites
+
+* You have a Red Hat account.
+* You have an offline token that authenticates the Terraform resources with the OpenShift Application Services API.
+
+[NOTE]
+====
+The offline token is a refresh token with no expiry and can be used by non-interactive processes to provide an access token for {product-long-rhoas} to authenticate the user. The token is an OpenShift offline token and you can find it at https://cloud.redhat.com/openshift/token. As the offline token is a sensitive value that varies between environments it is best specified as an `_OFFLINE_TOKEN_` environment variable when running `terraform apply` in a terminal. To set this environment variable, enter the following command in a terminal window, replacing <offline_token> with the value of the offline token:
+[source, subs="+quotes"]
+----
+export OFFLINE_TOKEN=<offline_token>
+----
+====
+
+.Procedure
+
+. Open the `main.tf` file in your IDE for editing.
+. Enter the example rhoas_kafka resource. This example uses the `"my-instance"`  identifier and creates a Kafka instance called `my-instance`. The only required fields are the `name`, `billing_model` and `plan` fields. The following values are set by default when you run `terraform apply`.
++
+.. `cloud provider`
+.. `region`
++
+All other information for the instance is provided by the Kafka APIs. The generated bootstrap server displays in the terminal as an output.
```

Comment:
If we add an explicit step for running `terraform apply`, then I think all of the following should follow that step as the result:

```
When you run the `terraform apply` command, Terraform automatically sets values for `cloud provider` and `region`.  All other information for the instance is provided by the Kafka APIs. The generated bootstrap server URL appears in the terminal as an output.
```